File: library-app/src/main/java/com/group/libraryapp/domain/user/loanhistory/UserLoanHistory.java
package com.group.libraryapp.domain.user.loanhistory;

import com.group.libraryapp.domain.user.User;
import org.jetbrains.annotations.NotNull;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

import static javax.persistence.GenerationType.IDENTITY;

@Entity
public class UserLoanHistory {

  @Id
  @GeneratedValue(strategy = IDENTITY)
  private Long id;

  @ManyToOne
  private User user;

  private String bookName;

  private boolean isReturn;

  public UserLoanHistory() {

  }

  public UserLoanHistory(User user, String bookName, boolean isReturn) {
    this.user = user;
    this.bookName = bookName;
    this.isReturn = isReturn;
  }

  @NotNull
  public String getBookName() {
    return this.bookName;
  }

  public void doReturn() {
    this.isReturn = true;
  }

  @NotNull
  public User getUser() {
    return user;
  }

  public boolean isReturn() {
    return isReturn;
  }
}

File: library-app/src/main/kotlin/com/group/libraryapp/domain/user/loanhistory/UserLoanHistory.kt
package com.group.libraryapp.domain.user.loanhistory

import com.group.libraryapp.domain.user.User
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.ManyToOne

@Entity
class UserLoanHistory(
        @ManyToOne
        val user: User,

        val bookName: String,

        var status: UserLoanStatus = UserLoanStatus.LOANED,

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        val id: Long? = null,
) {
    fun doReturn() {
        this.status = UserLoanStatus.RETURNED
    }

    companion object {
        fun fixture(
                user: User,
                bookName: String = "이상한 나라의 엘리스",
                status: UserLoanStatus = UserLoanStatus.LOANED,
                id: Long? = null,
        ): UserLoanHistory {
            return UserLoanHistory(
                    user = user,
                    bookName = bookName,
                    status = status,
                    id = id,
            )
        }
    }
}